code/smoothie.js

```diff
@@ -0,0 +1,314 @@
+  // }
+
+
+  // If a favorite is selected, remove the extra favorite-objects in array
+  const removeFavorites = () => {
+    if (mainFruitsPlusFavorite.length > 6) {
+      mainFruitsPlusFavorite.splice(-3, 3)
+    }
+  }
+
+  // If favorite is selected in browser, add matching object from mainFruit-array x 3 
+  // to a new array which will later be used to randomize fruit 
+  if (favoriteFruit == "Apple") {
+    removeFavorites()
+    mainFruitsPlusFavorite.push(mainFruits[0], mainFruits[0], mainFruits[0])
+  } else if (favoriteFruit == "Banana") {
+    removeFavorites()
+    mainFruitsPlusFavorite.push(mainFruits[1], mainFruits[1], mainFruits[1])
+  } else if (favoriteFruit == "Mango") {
+    removeFavorites()
+    mainFruitsPlusFavorite.push(mainFruits[2], mainFruits[2], mainFruits[2])
+  } else if (favoriteFruit == "Orange") {
+    removeFavorites()
+    mainFruitsPlusFavorite.push(mainFruits[3], mainFruits[3], mainFruits[3])
+  } else if (favoriteFruit == "Raspberry") {
+    removeFavorites()
+    mainFruitsPlusFavorite.push(mainFruits[4], mainFruits[4], mainFruits[4])
+  } else if (favoriteFruit == "Lemon") {
+    removeFavorites()
+    mainFruitsPlusFavorite.push(mainFruits[5], mainFruits[5], mainFruits[5])
+  }
+}
+
+// Function that selects a random object from chosen array
+const randomSelector = array => {
+  return array[Math.floor(Math.random() * array.length)]
+}
+
+// Function that toggles loading-gif
+const toggleLoader = (whatBox) => {
+  document.getElementById(whatBox).classList.add('loading')
+  // remove the loading-gif...
+  const removeLoader = () => {
+    document.getElementById(whatBox).classList.remove('loading')
+  }
+  //... after 1.5 second
+  setTimeout(removeLoader, 1500)
+}
+
+// 3 x Display random image and title from each category 
+// + add value to smoothie-object
+// + check if smoothie-button color should be changed
+const pickMainFruit = () => {
+  mainFruit = randomSelector(mainFruits)
+
+  // Pick a random fruit, unless favorite is selected 
+  // (then pick random from mainFruitsPlusFavorite-array)
+  if (mainFruitsPlusFavorite[6] == undefined) {
+    mainFruit = randomSelector(mainFruits)
+  } else {
+    mainFruit = randomSelector(mainFruitsPlusFavorite)
+  }
+
+  // Change titile of Main Fruit to reveal what the fruit is
+  const toggleMainFruitTitle = () => {
+    document.getElementById('fruit-header').innerHTML = `Main Fruit/berry: ${mainFruit.name}`
+  }
+
+  toggleLoader('main-fruit-box')
+  // Delay the reveal of name with same as loader
+  setTimeout(function () { toggleMainFruitTitle(); }, 1500)
+
+  document.getElementById('main-fruit-box').style.background = `url('${mainFruit.image}')`
+
+  smoothie.mainFruit = mainFruit.name
+  colorCheck()
+}
+
+const pickMixer = () => {
+  mixer = randomSelector(mixers)
+
+  // Change titile of Main Fruit to reveal what the fruit is
+  const toggleMixerTitle = () => {
+    document.getElementById('mixer-header').innerHTML = `Mixer: ${mixer.name}`
+  }
```

Comment:
Is it possible for these functions to be declared outside of this function?  And maybe send `mixer` as an argument?